pkg/utils/dataset_runtime.go

```diff
@@ -40,12 +43,24 @@ func GetRuntimeByCategory(runtimes []datav1alpha1.Runtime, category common.Categ
 // CreateRuntimeForReferenceDatasetIfNotExist creates runtime for ReferenceDataset
 func CreateRuntimeForReferenceDatasetIfNotExist(client client.Client, dataset *datav1alpha1.Dataset) (err error) {
 	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		_, err = GetThinRuntime(client,
+		runtime, err := GetThinRuntime(client,
 			dataset.GetName(),
 			dataset.GetNamespace())
 		// 1. if err is null, which indicates that the runtime exists, then return
 		if err == nil {
-			return nil
+			runtimeToUpdate := runtime.DeepCopy()
+			runtimeToUpdate.SetOwnerReferences([]metav1.OwnerReference{
+				{
+					Kind:       dataset.GetObjectKind().GroupVersionKind().Kind,
+					APIVersion: dataset.APIVersion,
+					Name:       dataset.GetName(),
+					UID:        dataset.GetUID(),
+					Controller: utilpointer.BoolPtr(true),
+				}})
+			if !reflect.DeepEqual(runtimeToUpdate, runtime) {
```

Comment:
I think the code may attempt to create runtime for multiple times: In cases that `runtimeToUpdate` is deep equal to `runtime`, the function will not return and go into the next `if IgnoreNotFound(err) == nil` statement.

Reply:
Thanks, I've added `return` in this `if` block.